ToolBox/RandomDataGenerator/RandomDataGeneratorProgram.cs

```diff
@@ -91,9 +91,12 @@ public static void GenerateRandomData(RandomDataGeneratorSettings settings, Cons
             foreach (var currentSymbol in symbolGenerator.GenerateRandomSymbols())
             {
                 // This is done so that we can update the symbol in the case of a rename event
+                TimeSpan span = settings.End.Subtract(settings.Start);
+                double diff_span = -(span.Minutes/2);
+                DateTime start_time = settings.End.AddMinutes(Math.Round(diff_span, 2, MidpointRounding.ToEven));
+                var delistDate = randomValueGenerator.NextDate(start_time, settings.End, null);
```

Comment:
Cool!
Think it would be great to add some unit tests around `GenerateRandomData`, it has none 💔, at least around this particular bug, the delisting date

Reply:
Added the unit test :)